src/modals/NewShift.vue

```diff
@@ -0,0 +1,461 @@
+          mode="tags"
+          placeholder="Select Manpower"
+          track-by="value"
+          label="value"
+          :close-on-select="false"
+          :options="tags"
+          @select="selectToggle"
+          @deselect="removeToggle"
+        />
+        <label>Manpower Detail</label>
+
+        <div v-if="selectedTags.length == 0"><br /></div>
+
+        <ol v-for="tag in selectedTags" :key="tag">
+          <input
+            class="manpower-qty"
+            type="number"
+            name="{{tag}}"
+            v-model="manpower[tag]"
+          />
+          <font-awesome-icon icon="fa-user" class="fa-user" />
+          <label for="{{tag}}"> x {{ tag }}</label>
+        </ol>
+        <div class="button-wrapper custom-action-row">
+          <button class="action-button done" type="button" @click="createShift">
+            Publish
+          </button>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
+import { db } from "@/firebase";
+import { collection, getDocs, addDoc } from "firebase/firestore";
+import Multiselect from "@vueform/multiselect/src/Multiselect";
+import dayjs from "dayjs";
+const dbTags = collection(db, "tags");
+
+export default {
+  name: "NewShift",
+
+  data() {
+    return {
+      repeating: false,
+      title: "",
+      date: "",
+      startTime: "",
+      endTime: "",
+      days: [
+        "Monday",
+        "Tuesday",
+        "Wednesday",
+        "Thursday",
+        "Friday",
+        "Saturday",
+        "Sunday",
+      ],
+      selectedDays: [],
+      tags: [],
+      selectedTags: [],
+      manpower: {
+        Clerk: 1,
+        Barista: 1,
+        Cashier: 1,
+      },
+      startDate: "",
+      endDate: "",
+      dayIndex: {
+        Monday: 1,
+        Tuesday: 2,
+        Wednesday: 3,
+        Thursday: 4,
+        Friday: 5,
+        Saturday: 6,
+        Sunday: 0,
+      },
+    };
+  },
+
+  props: ["branch"],
+  emit: ["togglePopup"],
+
+  components: { Multiselect },
+
+  async mounted() {
+    const tagsQuery = await getDocs(dbTags);
+    tagsQuery.forEach((doc) => {
+      this.tags.push(doc.data().tag);
+    });
+  },
+
+  methods: {
+    async createShift() {
+      // check if fields are filled up
+      if (this.title == "") {
+        alert("Please fill in the Title!");
+        return;
+      }
+      if (this.startTime == "") {
+        alert("Please fill in the Time In!");
+        return;
+      }
+      if (this.endStart == "") {
+        alert("Please fill in the Time Out!");
+        return;
+      }
+      if (
+        parseInt(this.endTime.replace(":", "")) -
+          parseInt(this.startTime.replace(":", "")) <=
+        0
+      ) {
+        alert("Your Time Out has to be after your Time In");
+        return;
+      }
+      if (this.selectedTags.length == 0) {
+        alert("Please select your Manpower!");
+        return;
+      }
+      if (
+        this.selectedTags.includes("Barista") &&
```

Comment:
Probably can refactor this logic, if any manpower is <=0 raise alert, cause there can be more tags than Barista Clerk and Cashier